### PR TITLE
add architecture/reapair/decommission docs

### PR DESCRIPTION
**Pull request by zmalik**
This PR adds documentation around already existing features like repair/decommission and also some architecture diagrams

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Apache Cassandra is a stateful workload. KUDO Cassandra uses kubernetes
+statefulset as the basic piece of the KUDO Cassandra Architecture
+
+As a StatefulSet maintains sticky identities for each of its Pods, this helps
+KUDO Cassandra to automate all necessary operations with Apache Cassandra nodes.
+
+To help with updates and upgrades, KUDO Cassandra comes with a custom config
+maps thats helps for rolling updates for KUDO Cassandra. Apache Cassandra
+maintenance jobs like `repair` and `backup/restore` are configured as kubernetes
+jobs and are only configured on-demand when configuring their respective
```
Comment (porridge) [inline docs/architecture.md:12]:
```suggestion
jobs and are only deployed on-demand when configuring their respective
```
configured ... configured does not sound nice

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+parameters.
+
+![](images/architecture.png)
+
+## Multi-Datacenter Architecture
+
+KUDO Cassandra can span a ring across multiple kubernetes clusters, to help the
```
Comment (porridge) [inline docs/architecture.md:19]:
```suggestion
KUDO Cassandra can span a ring across multiple kubernetes clusters, to facilitate the
```

File: docs/decommission.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Decommission KUDO Cassandra nodes
+
+KUDO Cassandra doesn't provide an automated way to scale down the Cassandra
+cluster, as this is a critical operation that shouldn't be repeated frequently.
+And to avoid KUDO Cassandra users falling in anti-patterns when managing an
```
Comment (porridge) [inline docs/decommission.md:5]:
```suggestion
KUDO Cassandra does not provide an automated way to scale down the Cassandra
cluster, as this is a critical operation that should not be repeated frequently,
and to discourage anti-patterns when managing an
```

File: docs/decommission.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Once the operation is completed, we can update the KUDO Cassandra Instance
+
+```
+kubectl kudo update -p NODE_COUNT=4 --instance analytics-cassandra -n dev
+```
+
+Once the update plan is complete, we can delete the PVC that was attached to the
+KUDO Cassandra `pod/analytics-cassandra-node-4`. Not deleting or cleaning the
+PVC will result in issues when scaling up the cluster next time.
```
Comment (porridge) [inline docs/decommission.md:43]:
```suggestion
PVC will result in issues when scaling the cluster up next time.
```

File: docs/repair.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Repair KUDO Cassandra
+
+KUDO Cassandra comes with a repair plan. Which can be triggered using
```
Comment (porridge) [inline docs/repair.md:3]:
```suggestion
KUDO Cassandra comes with a repair plan. It can be triggered using the
```

File: docs/repair.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+`REPAIR_POD` parameter Let's see with an example of a 3 node cluster
```
Comment (porridge) [inline docs/repair.md:4]:
```suggestion
`REPAIR_POD` parameter.

 Let's see with an example of a 3 node cluster
```

**Comment by zmalik** [comment]
@ANeumann82 @porridge I rebased branch with master and also made links in readme relative to fix links in https://github.com/kudobuilder/operators/tree/master/repository/cassandra/3.11

**Pull request merged**